### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by ViachaslauKabak**
Merged 2 roles into one
code refactoring

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
+Solr Configuration as Master or Slave
+=========
+[![License](https://img.shields.io/badge/license-Apache-green.svg?style=flat)](https://raw.githubusercontent.com/lean-delivery/ansible-role-solr-cluster/master/LICENSE)
+[![Build Status](https://travis-ci.org/lean-delivery/ansible-role-solr-cluster.svg?branch=master)](https://travis-ci.org/lean-delivery/ansible-role-solr-cluster)
+[![Build Status](https://gitlab.com/lean-delivery/ansible-role-solr-cluster/badges/master/build.svg)](https://gitlab.com/lean-delivery/ansible-role-solr-cluster)
```
Comment (tgadiev) [inline README.md]:
```suggestion
[![Build Status](https://gitlab.com/lean-delivery/ansible-role-solr-cluster/badges/master/build.svg)](https://gitlab.com/lean-delivery/ansible-role-solr-cluster/pipelines)
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  - **Supported OS**:
+    - CentOS
+      - 7
+    - Ubuntu
+    - Windows
+      - "Windows Server 2008"
+      - "Windows Server 2008 R2"
+      - "Windows Server 2012"
+      - "Windows Server 2012 R2"
+      - "Windows Server 2016"
```
Comment (tgadiev) [inline README.md:32]: Windows Server 2019?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+```
+
+Example Playbook
+----------------
+
+```yml
+- name: Configure Solr as slave
+  hosts: solr-slave
+  roles:
+    - role: lean_delivery.java
```
Comment (tgadiev) [inline README.md:153]: Does it work with default java settings?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    - role: lean_delivery.solr_standalone
+    - role: lean_delivery.solr_cluster
+      solr_instance_type: slave
+
+- name: Configure Solr as master
+  hosts: solr-master
+  roles:
+    - role: lean_delivery.java
```
Comment (tgadiev) [inline README.md:161]: Does it work with default java settings?

File: meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+---
+galaxy_info:
+  role_name: "solr-cluste"
```
Comment (tgadiev) [inline meta/main.yml]:
```suggestion
  role_name: "solr-cluster"
```

File: meta/main.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  author: "Lean Delivery team <team@lean-delivery.com>"
+  description: "Lean Delivery Solr configuration as master or slave"
+  company: "Epam Systems"
+  license: "Apache"
+  min_ansible_version: "2.7"
+  issue_tracker_url: "https://github.com/lean-delivery/ansible-role-solr-cluster/issues"
+  platforms:
+    - name: "Ubuntu"
+      versions:
+        - "xenial"
```
Comment (tgadiev) [inline meta/main.yml:13]: bionic?

File: tasks/system/Windows.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  when: auto_populate_configset_list
+
+- name: Populate configset list
+  set_fact:
+    configset_list: >-
+      {{ configset_output.files | map(attribute='path') | list }}
+  when: auto_populate_configset_list
+
+- name: Get config files list contained replication handler
+  win_command: >
```
Comment (tgadiev) [inline tasks/system/Windows.yml:16]: Why `win_command` instead of `win_find`?

File: tasks/system/Windows.yml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    regexp:
+      '<!--\sBEGIN\ssolr\sansible\scustom\sblock(.*)
+      END\ssolr\sansible\scustom\sblock\s-->'
+  when: item.stdout_lines | length
+  loop: '{{ true_configset_xml_list.results }}'
+  notify:
+    - restart Solr Windows
+
+- name: Force all notified handlers
+  meta: flush_handlers
```
Comment (tgadiev) [inline tasks/system/Windows.yml]: Why do we need to flush handlers explicitly?

File: tasks/system/Linux.yml [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    block: '{{ solr_config_lines }}'
+    state: present
+  become: True
+  loop: '{{ true_configset_xml_list.results }}'
+  when: item.matched
+  notify:
+    - restart Solr Linux
+
+- name: Force all notified handlers
+  meta: flush_handlers
```
Comment (tgadiev) [inline tasks/system/Linux.yml]: Why do we need to flush handlers explicitly?

**Review by tgadiev — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (ViachaslauKabak) [reply README.md:153]: yes

File: README.md (hunk 4 above)
Comment (ViachaslauKabak) [reply README.md:161]: yes

File: tasks/system/Windows.yml (hunk 7 above)
Comment (ViachaslauKabak) [reply tasks/system/Windows.yml:16]: win_find can examine file content?

File: tasks/system/Windows.yml (hunk 8 above)
Comment (ViachaslauKabak) [reply tasks/system/Windows.yml]: cause we need working solr till the end of this role

File: tasks/system/Linux.yml (hunk 9 above)
Comment (ViachaslauKabak) [reply tasks/system/Linux.yml]: cause we need working solr till the end of this role

**Review by tgadiev — APPROVED** [review]

**Pull request merged**